### PR TITLE
Jetpack Manage: Add user feedback modal form.

### PR DESCRIPTION
**Pull request by jkguidaven**
This pull request adds a modal form for user feedback. It can be accessed by clicking the **'Share user feedback'** option in the sidebar.

Screen Shot 2024-01-03 at 3 09 28 PM

**Note: The submit button is not functional and will be addressed in a separate PR.**

Closes https://github.com/Automattic/jetpack-genesis/issues/163

## Proposed Changes

* Implement the User feedback form modal based on the `Modal` component.
* Add 'Click' callback in the Share feedback button to show the modal.

## Testing Instructions
Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb.

* Use the Jetpack cloud live link below and go to the Dashboard page (`/dashboard?flags=jetpack/user-feedback-form`)
* Click the **'Share user feedback'** menu item in the sidebar.
screen-shot-2023-12-20-at-7 39 21-pm

* Confirm the modal shows up. Verify that the components within the modal work as expected.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/jetpack-cloud/components/user-feedback-form-modal/style.scss [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	padding: 24px 40px;
+	text-align: right;
+	background-color: var(--studio-gray-0);
+	border-radius: 4px;
+}
+
+.user-feedback-form-modal__close-button {
+	position: absolute;
+	right: 1rem;
+	top: 1rem;
```
Comment (yashwin) [inline client/jetpack-cloud/components/user-feedback-form-modal/style.scss]: Could we use RTL equivalent styles here?

```suggestion
	inset-inline-end: 1rem;
	inset-block-start: 1rem;
```

File: client/jetpack-cloud/components/user-feedback-form-modal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+type Props = {
+	show: boolean;
+	onClose: () => void;
+};
+
+export default function UserFeedbackForm( { show, onClose }: Props ) {
+	const translate = useTranslate();
+
+	const [ feedback, setFeedback ] = useState( '' );
+	const [ rating, setRating ] = useState( 3 );
```
Comment (yashwin) [inline client/jetpack-cloud/components/user-feedback-form-modal/index.tsx]: Is there a reason we are setting this to 3?

File: client/jetpack-cloud/components/user-feedback-form-modal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	}, [] );
+
+	const title = translate( 'Help us make Jetpack Manage even better' );
+
+	if ( ! show ) {
+		return null;
+	}
+
+	return (
+		<Modal className="user-feedback-form-modal" onRequestClose={ onClose } __experimentalHideHeader>
```
Comment (yashwin) [inline client/jetpack-cloud/components/user-feedback-form-modal/index.tsx]: Can we match the class names with the component name: `user-feedback-form`?

**Review by yashwin — COMMENTED** [review]
Overall looks good. I have left a few comments.

Also, a small clarification:

The state is not reset when I close the modal. Is this expected?

Screenshot 2024-01-03 at 2 44 28 PM

File: client/jetpack-cloud/components/user-feedback-form-modal/index.tsx (hunk 3 above)
Comment (jkguidaven) [reply client/jetpack-cloud/components/user-feedback-form-modal/index.tsx]: Good catch! I realized that this kind of component is commonly called a `ModalForm`. Therefore, I have renamed it to `UserFeedbackModalForm` instead.

File: client/jetpack-cloud/components/user-feedback-form-modal/index.tsx (hunk 2 above)
Comment (jkguidaven) [reply client/jetpack-cloud/components/user-feedback-form-modal/index.tsx]: I was initially considering setting this to a neutral value of 3 (not good or bad). However, upon reviewing how most apps do it, I realized they set it to 0. So thanks for raising this 👍

**Comment by jkguidaven** [comment]
> The state is not reset when I close the modal. Is this expected?

Yes, this will be fix on a separate PR that makes the modal URL accessible.

**Comment by jkguidaven** [comment]
@yashwin, thanks for the review. I made some adjustments based on your comment. Can you have another look? 🙏

**Review by yashwin — APPROVED** [review]
Thanks for addressing my comments. It works as expected, and code changes look reasonable to me. Good job, @jkguidaven!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10666017

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @jkguidaven for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10666017).

**Pull request merged**